### PR TITLE
filter out deleted and non-public add-ons by default from collection addon list api

### LINKED ISSUE #6845: Exclude deleted add-ons from collections endpoint by default
Currently the collections api endpoint includes deleted add-ons by default. 

See https://addons.mozilla.org/api/v3/accounts/account/mozilla/collections/featured-personas/addons/?sort=added for an example.

The suggested approach was to exclude deleted add-ons by default and have a separate param that can be used to request a collection including deleted add-ons for editing purposes. E.g. a `with_deleted` filter.

We should also consider how much the the `with_deleted` would be needed - although I can see that it might be handy to know that an add-on you added to a collection has now been deleted during editing.

### PR DESCRIPTION
**Pull request by eviljeff**
+ add filters to include them.

Fixes #6845 

Tbh, not 100% on the filter names.

### CODE REVIEW

File: docs/topics/api/collections.rst [hunk 1]
```diff
@@ -154,6 +155,22 @@ All sort parameters can be reversed, e.g. '-added' for descending dates.
...
+   By default, the collection addon list API will only return public add-ons
+   (excluding add-ons that have no approved listed versions, are disabled or
+   deleted) - you can change that with the ``filter`` query parameter:
+
+    =======================  ==================================================
+                      Value  Description
+    =======================  ==================================================
+                with_hidden  Show addons that have non-public statuses, but
+                             still exclude deleted add-ons.
+    with_hidden_and_deleted  Show all add-ons in the collection, regardless of
```
Comment (diox) [inline docs/topics/api/collections.rst]: Per IRC discussion: let's change that to `all` and `azll_with_deleted`

File: src/olympia/bandwagon/tests/test_views.py [hunk 2]
```diff
@@ -1809,6 +1823,22 @@ def test_sorting(self):
...
+        response = self.send(self.url + '?filter=with_hidden')
+        assert response.status_code == 200
+        # Now there should be 2 extra
+        assert len(response.data['results']) == 5
+
+        response = self.send(self.url + '?filter=with_hidden_and_deleted')
+        assert response.status_code == 200
+        # And one more still - with_deleted gets you with_hidden too.
+        assert len(response.data['results']) == 6
+
```
Comment (diox) [inline src/olympia/bandwagon/tests/test_views.py]: Can you add a quick test on the results to make sure we're returning the right ones ? No need to check for order, so just a `set()` with the ids is enough.

File: src/olympia/bandwagon/views.py [hunk 3]
```diff
@@ -710,5 +710,17 @@ def get_object(self):
...
     def get_queryset(self):
-        return CollectionAddon.objects.filter(
+        qs = CollectionAddon.objects.filter(
             collection=self.get_collection_viewset().get_object())
+        filter_param = self.request.GET.get('filter')
+        # We only filter list action.
+        include_deleted = (filter_param == 'with_hidden_and_deleted' or
+                           self.action != 'list')
+        # If deleted addons are requested, that implies hidden addons.
+        include_hidden = filter_param == 'with_hidden' or include_deleted
```
Comment (diox) [inline src/olympia/bandwagon/views.py]: Change the `include_*` variables to match the new filter names (see comment about the docs above)

File: src/olympia/bandwagon/tests/test_views.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/bandwagon/tests/test_views.py]: https://github.com/mozilla/addons-server/pull/6855/commits/9d1a6eec308638821a644fe6205ef954f0910df2

**Pull request merged**